TKeazirian.HTTPServer/Router/Router.cs

```diff
@@ -0,0 +1,92 @@
+namespace TKeazirian.HTTPServer.Router;
+
+using Handler;
+using Response;
+using Request;
+
+public class Router
+{
+    private readonly RoutesConfig _routesConfig;
+
+    public Router(RoutesConfig routesConfig)
+    {
+        _routesConfig = routesConfig;
+    }
+
+    public Response Route(Request request)
+    {
+        var handler = GetHandler(request);
+
+        if (RouteConfigContainsPath(request) && !IsHttpMethodAllowed(request))
+        {
+            return new NotImplementedResponse().BuildNotImplementedResponse();
+        }
+
+        if (RouteConfigContainsPath(request) && IsHttpMethodAllowed(request))
+        {
+            if (IsHeadRequest(request, handler))
+            {
+                return new HeadResponse().BuildHeadResponse(handler, request);
+            }
+
+            if (IsOptionsResponse(request, handler))
+            {
+                return new OptionsResponse().BuildOptionsResponse(request, _routesConfig);
+            }
+        }
+
+        return handler.HandleResponse(request);
```

Comment:
@Aoxian I need it here too ([see](https://github.com/t-keazirian/HTTPServer/pull/27/files/afdccc7a4c47817bf115dc624d69f734f543575f#r915089579))